src/arithmetic/arithmetic_expression_construct.c

```diff
@@ -441,6 +443,96 @@ static AR_ExpNode *_AR_ExpFromNamedPath(const cypher_astnode_t *path) {
 	return op;
 }
 
+static AR_ExpNode *_AR_ExpFromShortestPath(const cypher_astnode_t *path) {
+	uint path_len = cypher_ast_pattern_path_nelements(path);
+	if(path_len != 3) {
+		ErrorCtx_SetError("shortestPath requires a path containing a single relationship");
+		return AR_EXP_NewConstOperandNode(SI_NullVal());
+	}
+
+	// Retrieve the minimum and maximum number of hops, if specified.
+	int start = 1;
+	int end = EDGE_LENGTH_INF;
+	const cypher_astnode_t *edge = cypher_ast_pattern_path_get_element(path, 1);
+	const cypher_astnode_t *range = cypher_ast_rel_pattern_get_varlength(edge);
+	if(range == NULL) {
+		end = 1; // Not a variable-length edge
+	} else {
+		const cypher_astnode_t *range_start = cypher_ast_range_get_start(range);
+		if(range_start) {
+			// If specified, the edge's minimum hop value must be 0 or 1
+			start = AST_ParseIntegerNode(range_start);
+			if(start > 1) {
+				ErrorCtx_SetError("shortestPath does not support a minimal length different from 0 or 1");
+				return AR_EXP_NewConstOperandNode(SI_NullVal());
+			}
+		}
+		const cypher_astnode_t *range_end = cypher_ast_range_get_end(range);
+		if(range_end) end = AST_ParseIntegerNode(range_end);
+	}
+
+	if(!cypher_ast_shortest_path_is_single(path)) {
+		ErrorCtx_SetError("RedisGraph does not currently support allShortestPaths");
+		return AR_EXP_NewConstOperandNode(SI_NullVal());
+	}
+
+	enum cypher_rel_direction dir = cypher_ast_rel_pattern_get_direction(edge);
+	if(dir == CYPHER_REL_BIDIRECTIONAL) {
+		ErrorCtx_SetError("RedisGraph does not currently support undirected shortestPath traversals");
+		return AR_EXP_NewConstOperandNode(SI_NullVal());
+	}
+
+	if(cypher_ast_rel_pattern_get_properties(edge)) {
```

Comment:
What about filters on the src and dest nodes?
`RETURN shortestPath ( (a{v:1})-[*]->(b{v:2}) )` ?

Reply:
Can you please add a test covering this?